File: atcoder.jp/contests/abc281/tasks/abc281_b/main.ts
import { readFileSync } from "fs";
let inputLinesIndex = 0;
const inputLines = readFileSync("/dev/stdin", "utf8").split("\n");
const input = () => inputLines[inputLinesIndex++];

const isUpper = (s: string) => isNaN(+s) && s.toUpperCase() === s;

const solve = (s: string) => {
  if (s.length !== 8) return false;
  if (!isUpper(s[0])) return false;
  if (!isUpper(s.slice(-1))) return false;
  const n = +s.slice(1, -1);
  return 100000 <= n && n <= 999999;
};

const main = () => {
  console.log(solve(input()) ? "Yes" : "No");
};

main();

File: atcoder.jp/contests/abc281/tasks/abc281_b/main.py
def cond2(s: str) -> bool:
    if 6 < len(s):
        return False
    try:
        return 100000 <= int(s) <= 999999
    except ValueError:
        return False


S = input()

ans = all([S[0].isupper(), cond2(S[1:-1]), S[-1].isupper()])
print("Yes" if ans else "No")
